File: B2C2C_Back_end/dockerized.postgresql/src/main/java/com/TFG_backend/dockerized/postgresql/invitacion/InvitacionRepository.java
package com.TFG_backend.dockerized.postgresql.invitacion;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

public interface InvitacionRepository extends JpaRepository<Invitacion, Long> {
    List<Invitacion> findByEmpresaId(Long empresaId);
    List<Invitacion> findByAlumnoId(Long alumnoId);
    List<Invitacion> findByOfertaId(Long ofertaId);
}

File: B2C2C_Front_end/app/src/main/java/com/torre/b2c2c_tfg/domain/repository/InvitacionRepository.kt
package com.torre.b2c2c_tfg.domain.repository

import com.torre.b2c2c_tfg.data.model.Invitacion

interface InvitacionRepository {
    suspend fun crearInvitacion(invitacion: Invitacion): Boolean
    suspend fun getInvitacionesByEmpresaId(empresaId: Long): List<Invitacion>
}